Controller/RestController.php

```diff
@@ -37,4 +41,107 @@ public function listAction()
 
         return new JsonResponse(array_combine($referenceDataNames, $referenceDataNames));
     }
+
+    /**
+     * Create a custom entity
+     *
+     * @param Request $request
+     * @param string  $customEntityName
+     *
+     * @return JsonResponse
+     */
+    public function createAction(Request $request, string $customEntityName): JsonResponse
```

Comment:
`RequestStack` instead

Reply:
I created it again as my comments seems done. Please use dedicated actions. It does not matter you're using JsonReponse ^^

Reply:
And you also will be able to remove all your protected methods which depend of the action you call and not of the class